packages/mint-components/src/components/sqm-rewards-table/cells/sqm-rewards-table-source-cell.tsx

```diff
@@ -0,0 +1,119 @@
+import { Component, h, Prop } from "@stencil/core";
+import { intl } from "../../../global/global";
+import { TextSpanView } from "../../sqm-text-span/sqm-text-span-view";
+
+@Component({
+  tag: "sqm-rewards-table-source-cell",
+  shadow: true,
+})
+export class RewardTableSourceCell {
+  @Prop() reward: Reward;
+  @Prop() deletedUserText: string;
+  @Prop() anonymousUserText: string;
+  @Prop() rewardExchangeText: string;
+  @Prop() referralText: string;
+  @Prop() rewardSourceText: string;
+  @Prop() locale: string = "en";
+
+  render() {
+    intl.locale = this.locale;
+    const RewardSource = ({ reward }: { reward: Reward }) => {
+      return (
+        <span>
+          {intl.formatMessage(
+            {
+              id: "rewardSourceText",
+              defaultMessage: this.rewardSourceText,
+            },
+            {
+              rewardSource: reward.rewardSource,
+              programName: reward?.program?.name,
+            }
+          )}
+        </span>
+      );
+    };
+
+    const SOURCE_COLUMN_LENGTH = 21;
+    const RewardExchangeBadge = ({ reward }: { reward: Reward }) => {
+      const rewardExchange = (
+        <div>
+          <span
+            style={{
+              fontSize: "var(--sl-font-size-small)",
+              color: "var(--sl-color-neutral-500)",
+            }}
+          >
+            {this.rewardExchangeText}
+          </span>
+          <br />
+          {reward.exchangedRewardRedemptionTransaction?.prettyRedeemedCredit}
+          {" → "}
+          {reward.prettyValue}
+        </div>
+      );
+
+      return reward.exchangedRewardRedemptionTransaction.prettyRedeemedCredit
+        .length +
+        reward.prettyValue.length <
+        SOURCE_COLUMN_LENGTH ? (
+        rewardExchange
+      ) : (
+        <div
+          style={{
+            display: "inline-block",
+            verticalAlign: "top",
+            // maxWidth: "155px",
+          }}
+        >
+          {rewardExchange}
+        </div>
+      );
+    };
+
+    // TODO: user type
```

Comment:
Flagging TODO for any type - easy fix

Reply:
Added `user: { firstName: string; lastName: string }`